Dockerfile

```diff
@@ -9,19 +9,36 @@ WORKDIR /var/cache
 RUN git clone https://github.com/SUSE/telemetry
 RUN cd telemetry; make build
 
-# Copy local code to the container image.
-RUN mkdir ./telemetry-server
+# Create dest directory for local code
+RUN mkdir -p ./telemetry-server/server/telemetry-server
+
+# Copy main go.mod and go.sum to dest directory and run go mod download
+COPY go.mod ./telemetry-server
+COPY go.sum ./telemetry-server
+RUN cd telemetry-server; go mod download
```

Comment:
Possibly, will explore more in later investigations...

This was primarily to reduce layer churn, as this way the build step is just doing a build using already downloaded modules, and if the modules haven't changed then the mod download step can use the cached layer... Otherwise the build step ends up downloading all the modules every time, which can be slow.